pkg/internal/ebpf/common/pids.go

```diff
@@ -81,25 +67,23 @@ func (pf *PIDsFilter) Filter(inputSpans []request.Span) []request.Span {
 	outputSpans := make([]request.Span, 0, len(inputSpans))
 	pf.updatePIDs()
 	for i := range inputSpans {
-		// While BPF always see the processes in the same way (from the kernel-side)
-		// Beyla userspace might see them differently depending on how it is operating.
-		// If they are in different namespaces, the process finder will
-		var pidView uint32
-		if inputSpans[i].Pid.Namespace == pf.pidNs {
```

Comment:
I might be missing something but I think if we remove this kind of comparison, I don't see how can it work for all the scenarios.

The reason to do this comparison was that we need to match the pid that we trace at the eBPF side with [the PID that Beyla reads when it discovers a new process that might be instrumentable](https://github.com/grafana/beyla/blob/main/pkg/internal/discover/watcher.go#L183).

Since this process PID is read at the user space (file `watcher.go`), its value will depend on where is Beyla located. If it runs as a sidecar container, the `watcher.go`'s `fetchProcessPorts` function will get the container-namespaced PIDs; if Beyla runs as a host process (or as a `hostPID: true` pod), it will get the host PIDs.

That's why we compare the Beyla namespace with the PID namespace: if both Beyla and the process are in the same namespace  compare the user namespace. If they are in different namespaces, compare the host namespace. And this fails when the container is in another Namespace that it is neither the Beyla userspace namespace, nor the host namespace (e.g. in Kind on top of Mac virtualized solutions).

To fix that third scenario, i'd try to (and I don't know whether this is possible) to know from the Beyla side all the possible `ns:pid` pairs for that process. If only one of them matches with the `ns:pid` as reported from the BPF side, then the process should be allowed.